Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -4754,8 +4754,55 @@ else if (currentLocationKey.getId()==pathLocation2Key.getId())
 
 		if (destination.getProperty("ownerKey")!=null)
 		{
-			if (isInParty)
+			if (isInParty) {
+				List<CachedEntity> partyMembers = getParty(db, character);
+				List<CachedEntity> partyGroups = new ArrayList<CachedEntity>();
+				List<CachedEntity> partyUsers = new ArrayList<CachedEntity>();
+				for(CachedEntity partyMember: partyMembers) {
+					//Removes those who have just applied
+					String groupStatus = (String)character.getProperty("groupStatus");
+					if(("Member".equals(groupStatus)==false && "Admin".equals(groupStatus)==false)) {
+						partyGroups.add(new CachedEntity());
+					} else {
+						partyGroups.add(partyMember.getProperty("groupKey"));
+					}
+					partyUsers.add(partyMember.getProperty("userKey"));
+				}
+				Key ownerKey = (Key)destination.getProperty("ownerKey");
+				if(ownerKey.getKind().equals("Group")){
+					//iterates through the list of groups that the party members belong to.
+					//if there are any differences, set the flag to false
+					boolean allInSameGroup = true;
+					for(int i = 0; i < partyGroup.size() - 1; i++) {
+						if(!partyGroups.get(i).equals(partyGroups.get(i+1))){
+							allInSameGroup = false;
+						}
+					}
+					if(allInSameGroup && ownerKey.equals(partyGroups.get(0))) {
+						return destination;
```

Comment:
Yes, this is intentional.  We already know the destination is someplace that has an owner, and we have done all of the checks that they should make to be able to enter, so returning here should be fine.